### PR TITLE
Problem: native fee handler don't respect DefaultPriorityReduction

### PR DESCRIPTION
**Pull request by yihuang**
- vendor fee deduct decorator, so it'll calls the custom `checkTxFeeWithValidatorMinGasPrices` method.
- a little refactor to remove `DeductTxCostsFromUserBalance` and call the DeductFees directly

next: will vendor the `DeductFees` method and modify the impl for parallel exec

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/handler_test.go [hunk 1]
```diff
@@ -500,7 +503,13 @@ func (suite *HandlerTestSuite) TestERC20TransferReverted() {
 			suite.Require().NoError(err)
 			fees, err := keeper.VerifyFee(txData, "aphoton", baseFee, true, true, true, suite.Ctx.IsCheckTx())
 			suite.Require().NoError(err)
-			err = k.DeductTxCostsFromUserBalance(suite.Ctx, fees, tx.GetSender())
+
+			// fetch sender account
+			signerAcc := suite.app.AccountKeeper.GetAccount(suite.ctx, tx.GetSender().Bytes())
```
Comment (yihuang) [inline x/evm/handler_test.go]:
```suggestion
			signerAcc := suite.App.AccountKeeper.GetAccount(suite.Ctx, tx.GetSender().Bytes())
```

File: x/evm/handler_test.go [hunk 2]
```diff
@@ -500,7 +503,13 @@ func (suite *HandlerTestSuite) TestERC20TransferReverted() {
...
+			suite.Require().NotNil(signerAcc)
+
+			// deduct the full gas cost from the user balance
+			err = ante.DeductFees(suite.app.BankKeeper, suite.ctx, signerAcc, fees)
```
Comment (yihuang) [inline x/evm/handler_test.go]:
```suggestion
			err = ante.DeductFees(suite.App.BankKeeper, suite.Ctx, signerAcc, fees)
```

**Comment by mmsqe** [comment]
seems fetch sender account not work as expected now

**Comment by yihuang** [comment]
> seems fetch sender account not work as expected now

WDYM? test passed now.

File: x/evm/keeper/utils_test.go [hunk 3]
```diff
@@ -531,7 +532,12 @@ func (suite *UtilsTestSuite) TestVerifyFeeAndDeductTxCostsFromUserBalance() {
 				suite.Require().Nil(fees, "invalid test %d passed. fees value must be nil - '%s'", i, tc.name)
 			}
 
-			err = suite.App.EvmKeeper.DeductTxCostsFromUserBalance(suite.Ctx, fees, common.BytesToAddress(tx.From))
+			// fetch sender account
+			acc := suite.App.AccountKeeper.GetAccount(suite.Ctx, tx.From)
```
Comment (mmsqe) [inline x/evm/keeper/utils_test.go:536]: need check if expectPassVerify invalid account

**Pull request closed without merging**